lib/Onnxifi/Base.cpp

```diff
@@ -141,36 +139,32 @@ onnxStatus Graph::setIOAndRun(uint32_t inputsCount,
       return ONNXIFI_STATUS_INVALID_SHAPE;
     }
 
-    // Only re-allocate a tensor in case padding is required.
-    // Otherwise just back the tensor by memory provided by the caller.
+    // Only allocate a tensor if insufficient backing storage is provided.
+    unsigned elementSize = inPhPtr->getType()->getElementSize();
+    size_t onnxBytes = inOnnxTensorSize * elementSize;
     if (inPhPtr->dims().equals(inOnnxTensorDims)) {
       ctx->getPlaceholderBindings()->insert(
-          inPhPtr, new Tensor(inOnnxBuffer, inPhPtr->getType()));
+          inPhPtr, Tensor(inOnnxBuffer, inPhPtr->getType()));
+    } else if (backendPtr_->getBackendId()
+                   ->getBackend()
+                   .supportsPartialTensors() &&
+               inOnnxBuffer && inOnnxTensorSize > 0) {
+      // We have a partial input buffer.  Create a padded unownded tensor that
+      // remembers the actual size of the input.
+      ctx->getPlaceholderBindings()->insert(
+          inPhPtr, Tensor(inOnnxBuffer, inPhPtr->getType(), onnxBytes));
     } else {
+      // If input onnxTensorDescriptor has a NULL buffer pointer, which is a
+      // valid case for empty tensor, skip copying.
       Tensor *inputTensor = tensorPool_.get(inPhPtr->getType());
       DCHECK(inputTensor) << "Tensorpool tensor not found for input "
```

Comment:
since this is only debug check we'd need to handle nullptr in inputTensor gracefully, otherwise we'd get segfault on line 163.
We should return appropriate ONNXIFI return code in case of no available tensors in a pool.

Reply:
I'm never sure whether we should check for programming errors at runtime in release builds.  Is it actually possible to get a nullptr back from `tensorPool_` in the absence of bugs?

Reply:
Theres a configuration option for the TensorPool where it wont allocate new tensors and will return null. We don't use it in Onnxifi though so this should always be valid.